components/aws/sagemaker/common/_utils.py

```diff
@@ -228,21 +244,73 @@ def create_training_job(client, args):
       raise Exception(e.response['Error']['Message'])
 
 
-def wait_for_training_job(client, training_job_name, poll_interval=30):
-  while(True):
-    response = client.describe_training_job(TrainingJobName=training_job_name)
-    status = response['TrainingJobStatus']
-    if status == 'Completed':
-      logging.info("Training job ended with status: " + status)
-      break
-    if status == 'Failed':
-      message = response['FailureReason']
-      logging.info('Training failed with the following error: {}'.format(message))
-      raise Exception('Training job failed')
-    logging.info("Training job is still in status: " + status)
-    time.sleep(poll_interval)
+def wait_for_training_job(client, training_job_name, poll_interval=31):
+    while(True):
+        response = client.describe_training_job(TrainingJobName=training_job_name)
+        status = response['TrainingJobStatus']
+        if status == 'Completed':
+            logging.info("Training job ended with status: " + status)
+            break
+        if status == 'Failed':
+            message = response['FailureReason']
+            logging.info('Training failed with the following error: {}'.format(message))
+            raise Exception('Training job failed')
+        logging.info("Training job is still in status: " + status)
+        time.sleep(poll_interval)
+
+
+def wait_for_debug_rules(client, training_job_name, poll_interval=31):
+    first_poll = True
+    while(True):
+        response = client.describe_training_job(TrainingJobName=training_job_name)
+        if 'DebugRuleEvaluationStatuses' not in response:
+            break
+        if first_poll:
+            logging.info("Polling for status of all debug rules:")
+            first_poll = False
+        if debug_rules_completed(response):
+            logging.info("Rules have ended with status:\n")
+            print_debug_rule_status(response, True)
+            break
+        print_debug_rule_status(response)
+        time.sleep(poll_interval)
 
 
+def debug_rules_errored(response):
+    if 'DebugRuleEvaluationStatuses' in response:
+        for debug_rule in response['DebugRuleEvaluationStatuses']:
+            if debug_rule['RuleEvaluationStatus'] == "Error":
+                return True
+    return False
+
+
+def debug_rules_completed(response):
+    if 'DebugRuleEvaluationStatuses' in response:
+        for debug_rule in response['DebugRuleEvaluationStatuses']:
+            if debug_rule['RuleEvaluationStatus'] == "InProgress":
+                return False
+    return True
+
+
+def print_debug_rule_status(response, last_print=False):
```

Comment:
The printing in this method is difficult to follow. If you wouldn't mind adding a docstring with an example of the output, that would make it much simpler to read through.

Reply:
Also would you mind clarifying what `last_print` means? Even when looking at the code directly it's unclear when I would use it.

Reply:
`last_print ` means all the debug rules have finished and we want to print it for the last time with the status details as well as a new line between each rule.